docs: add changelog updates (#563)
fleek-platform/website: src/content/changelog/2025-changelog-eliza-ssh-access-for-agents-api-key-generation-and-more/index.md

---
title: 'Fleek changelog: Version toggling, API key generation, and more'
date: 2025-03-03

> desc: 'Eliza version selector for AI agent deployment, API key generation, and more'

Reviewer: The description seem to be repeating what's already described in the title.

```suggestion
desc: 'Latest update brings Eliza version selection functionality for AI agent deployments, an API key generation tool, and various platform optimizations.'
```

thumbnail: './fleek-changelog.png'
image: './fleek-changelog.png'
author:
  - 'Tobiloba Adedeji'
---

Hey, Fleek community!

We’ve rolled out some exciting updates over the past few days, and we’re excited to share them with you! Here’s a breakdown of what’s new:

---

## Eliza Version Control

Fleek allows users to switch between different versions of Eliza while building AI agents. Through the **Agent Deployment Portal**, users can select a specific Eliza version and adapt their agent’s instance on Fleek’s virtual machines.

> ### Supported Versions

Reviewer: ```suggestion
### Supported versions
```

Fleek currently supports [Eliza v0.25.9](https://elizaos.github.io/eliza/docs/changelog/), [Eliza v0.1.9](https://elizaos.github.io/eliza/docs/changelog/#v019-2025-02-01), [Eliza v0.1.8](https://elizaos.github.io/eliza/docs/changelog/#v018-alpha1-january-31-2025) and [Eliza v0.1.7alpha2](https://elizaos.github.io/eliza/docs/changelog/#version-017-alpha2-2024-12-28). You can track updates and available versions [here](https://elizaos.github.io/eliza/docs/changelog/).

> ### Switching Eliza Versions
> 
> To switch between Eliza versions:
> 
> 1. Visit [fleek.xyz/eliza/](https://fleek.xyz/eliza/).
> 2. Use the **dropdown toggle** to select the desired version.
> 3. Proceed with one of the [three deployment options](/docs/ai-agents/#deploy-ai-agents) on the Deployment Portal.
> 
> This flexibility allows developers to test, iterate, and deploy AI agents on the latest stable versions while maintaining compatibility.

Reviewer: We can maybe be a bit more organic with the versioning, as "maintaining compatibility" is not guaranteed to predirect future versions of Eliza? So, we can underline versioning only for now, something like:

```suggestion
### Switching between Eliza versions

When preparing to deploy an AI agent on Fleek, you can select the Eliza version by:

1. Visiting [fleek.xyz/eliza/](https://fleek.xyz/eliza/).
2. Using the **drop-down toggle** to select the desired version.
3. Proceeding with one of the [three deployment options](/docs/ai-agents/#deploy-ai-agents)

This provides greater control over testing, iterating, and deploying AI agents on the latest stable versions.
```

---

## API Key Generation Made Easier

> Generating API tokens for Fleek’s [AI Agents APIs](/docs/ai-agents/agents-apis/) is now more streamlined. You can now create, view, copy, and regenerate API keys directly from the [Eliza deployment page](/eliza/) or the [Fleek dashboard](https://app.fleek.xyz/).

Reviewer: Let's reduce the number of "now"

```suggestion
Generating API tokens for Fleek’s [AI agents APIs](/docs/ai-agents/agents-apis/) is now more streamlined. You can create, view, copy, and regenerate API keys directly from the [Eliza deployment page](/eliza/) or the [Fleek dashboard](https://app.fleek.xyz/).
```

**Steps to generate an API key:**

1. Visit the [Eliza deployment page](https://fleek.xyz/eliza) and click **"Account settings"** at the bottom.

> 2. Navigate to the **"API Tokens"** tab to view all your generated API keys.

Reviewer: ```suggestion
2. Navigate to the **API Tokens** tab to view all your generated API keys.
```

> 3. Click **"Create API Token". A** modal appears, \*\*\*\*enter a name for your key and confirm.

Reviewer: Modal is more of a developer terminology. Let's call it pop-up to be a bit friendlier for general users.

```suggestion
3. Click **Create API Token**. A pop-up should show up requesting to enter a name for your key and a confirmation.
```

> ![Create API Token](./create-api-token.png)

Reviewer: ```suggestion
![Create an API Token](./create-api-token.png)
```

Use the generated key as your X-Api-Key for all API requests.

---

> ## AI Agent Draft Status

Reviewer: ```suggestion
## AI agent draft status
```

> Agents still in draft mode will feature an icon next to their name on the Eliza deployment page and the Fleek dashboard.

Reviewer: ```suggestion
Draft agents' status will be reflected in an icon next to their name on the Eliza deployment page and the Fleek dashboard.
```

Additionally, users can now:

> - Easily view them with all the other agents

Reviewer: You mean that the user can now see the status for any listed agent right? So, maybe:

```suggestion
- Easily confirm the status of any listed agent 
```

> - Continue editing drafts before publishing

Reviewer: ```suggestion
- Edit AI agents in drafts mode before publishing
```

- Delete draft agents if no longer needed

> This refinement streamlines agent development and ensures drafts don’t get lost in larger projects.

Reviewer: ```suggestion
Quickly browse all agents through a simplified list view, reducing any chances of missing any in larger projects.
```

> The draft creation APIs are also available now. Learn more in our [AI Agents API documentation](/docs/ai-agents/agents-apis/).

Reviewer: ```suggestion
The draft creation APIs are also available now. Learn more in our [AI agents API documentation](/docs/ai-agents/agents-apis/).
```

![Draft AI agent](./draft-agent.png)

---

> ## Table of Contents for AI Agent Creation

Reviewer: ```suggestion
## Table of contents for AI agent creation
```

The AI agent creation flow for both "Build from Scratch" and "Start with a Template" now includes a table of contents for seamless navigation. You can quickly jump between sections to review and edit details before proceeding to deployment.

![Table of contents](./templates.png)

---

> ## Sidebar Status Indicators for AI Agents

Reviewer: Hasn't this been described earlier? I guess you can keep the title shorter, as the status is available everywhere the avatar's displayed, including the sidebar you want to illustrate here, etc.

```suggestion
## AI agents status
```

> The AI Agents sidebar on the [Eliza deployment page](/eliza) now displays agent statuses at a glance. This helps you quickly identify which agents are active, inactive, or still in draft mode.

Reviewer: ```suggestion
Now you can instantly see which agents are active, inactive, or in draft mode with the updated AI agent avatar component, displayed in the sidebar, settings panel, etc.
```

> ### Updated Sidebar Indicators

Reviewer: Seems that we have at least 3 topics for status. Can't it be condensed into a single one? For example, do we really need to highlight "Updated sidebar indicators" when status is a new feature, so an update over indicator doesn't make sense? Am I missing something here?

I'd advise simplifying all of it in a single topic please.

### Switching between Eliza versions

When preparing to deploy an AI agent on Fleek, you can select the Eliza version by:

1. Visiting [fleek.xyz/eliza/](https://fleek.xyz/eliza/).
2. Using the **drop-down toggle** to select the desired version.
3. Proceeding with one of the [three deployment options](/docs/ai-agents/#deploy-ai-agents).

This provides greater control over testing, iterating, and deploying AI agents on the latest stable versions.

---

## API key generation made easier

Generating API tokens for Fleek’s [AI agents APIs](/docs/ai-agents/agents-apis/) is now more streamlined. You can create, view, copy, and regenerate API keys directly from the [Eliza deployment page](/eliza/) or the [Fleek dashboard](https://app.fleek.xyz/).

**Steps to generate an API key:**

> 1. Visit the [Eliza deployment page](https://fleek.xyz/eliza) and click **"Account settings"** at the bottom.

Reviewer: ```suggestion
1. Visit the [Eliza deployment page](https://fleek.xyz/eliza) and click **Account settings** at the bottom.
```

2. Navigate to the **API Tokens** tab to view all your generated API keys.
3. Click **Create API Token**. A pop-up should show up requesting to enter a name for your key and a confirmation.

![Create an API Token](./create-api-token.png)

You can now apply the generated key as your `X-Api-Key` request header when making API requests.

---

> ## AI agent status and management

Reviewer: I'd think that management typically encompasses status monitoring, etc. So, maybe we can make the title:

```suggestion
## AI agent management improvements
```

> We've made significant improvements to AI agent status tracking and management, making it easier to monitor, edit, and organize your agents efficiently.

Reviewer: To be more in tune with our title we can do:

```suggestion
We've made significant improvements to AI agent management, including status monitoring. It's easier to monitor, edit, and organize your agents efficiently.
```
